src/Client.php

```diff
@@ -183,9 +176,14 @@ protected function getApiContentType()
      */
     public function getAsset($assetId, $locale = null)
     {
-        $locale = null === $locale ? $this->defaultLocale : $locale;
+        $locale = $locale ?: $this->defaultLocale;
 
-        return $this->requestAndBuild('GET', 'assets/'.$assetId, [
+        $instanceId = $assetId.'-'.($locale ?: '*');
+        if ($this->instanceRepository->has('Asset', $instanceId)) {
+            return $this->instanceRepository->get('Asset', $instanceId);
+        }
+
+        return $this->requestAndBuild('/spaces/'.$this->spaceId.'/assets/'.$assetId, [
```

Comment:
The `'/spaces'.$this->spaceID` could be extracted to the `requestAndBuild` method, as there is no endpoint in the CDA that doesn't require that path.

Reply:
No, I actually want to have the full URL. Previously it would set the `/spaces/.$this->spaceId` as part of the HTTP client base URI options, but I changed that behavior because I'd rather have consistency than save a few characters.